plugins/nodes/src/js/columns/NodesTableCPUColumn.tsx

```diff
@@ -1,24 +1,34 @@
 import * as React from "react";
-import { compareValues } from "#PLUGINS/nodes/src/js/utils/compareValues";
+import { sort, compareAsNumber, compareAsString } from "@dcos/sorting";
 import Node from "#SRC/js/structs/Node";
 // TODO: DCOS-39079
 // import { IWidthArgs as WidthArgs } from "@dcos/ui-kit/packages/table/components/Column";
 import { IWidthArgs as WidthArgs } from "#PLUGINS/nodes/src/js/types/IWidthArgs";
-import { SortDirection } from "plugins/nodes/src/js/types/SortDirection";
+import {
+  SortDirection,
+  directionAwareComparators
+} from "../types/SortDirection";
 
 export function cpuRenderer(data: Node): React.ReactNode {
   return `${data.getUsageStats("cpus").percentage}%`;
 }
+
+function getCpuUsage(node: Node) {
+  return node.getUsageStats("cpus").percentage;
+}
+
+function getHostName(node: Node) {
+  return node.getHostName().toLowerCase();
+}
+
 export function cpuSorter(data: Node[], sortDirection: SortDirection): Node[] {
-  const sortedData = data.sort((a, b) =>
-    compareValues(
-      a.get("used_resources").cpus.toString(),
-      b.get("used_resources").cpus.toString(),
-      a.getHostName().toLowerCase(),
-      b.getHostName().toLowerCase()
-    )
+  return sort(
```

Comment:
Here were are tieing sort and compare together, in a way that the cpuComparator done here is not reusable in other parts of the system. Have you considered moving the lenses and the comparator closer to the type, or at least out of the column?

Reply:
I decided to not move it somewhere else as it handles the sorting for this column. I don't want to have a global "this sorts by CPU" functionality, because I think it is prone to be extended to 🍝 code. It's also not so much code that I really see a point in drying it up yet, but feel free to convince me :)

Reply:
I would be ok with a private function on this module.

Reply:
I can't follow you. Do you mean typescript private or something like `_sortByHealth`? What is the problem with making this public?

Reply:
```ts
sort(cpuComparator(sortDirection), data)

function cpuComparator(sortDirection): Comparator<Node> {
  return directionAwareComparators(
      [compareAsNumber(getCpuUsage), compareAsString(getHostName)],
      sortDirection
    );
}
```

My proposal is to separate comparison, from sorting into its own functions. What do you think of this code? You can leave it here, but you can also move somewhere else if the need to compare emerges as part of the type somewhere else.

Reply:
I don't really understand why this is significantly better to you or why it matters to you in general.

Reply:
Separate comparison from sort, which are for me different concerns. I would go as to say that running the sort could even be in the table, and not in the column. 

If you think on the Strategy pattern terms, this function couple Strategy and Algorithm tightly so Strategy is not reusable.

But on any case, I don't really think it is important, so if you still disagree lets just keep it this way.

Reply:
I see your point with having the sort on the table, I would be completely fine with just exporting a set of comparators and that's it :)

Can you explain to me why it couples strategy and algorithm? We have the sorting algorithms split out, the only "algorithm" we still have is for extracting data from the node struct as far as I understand it.

Thank you for explaining it again :)

Reply:
Sorry, missed that one. Sort, Min, Max are all algorithms in the sense they process a collection and return some meaningful information. But compareBy is the strategy they use to achieve their result. They usually have an implicit strategy supported, because they are used all the time, but all those 3 algorithms are equally configurable by passing them a comparison strategy.

```js
sort(["a","b","c"], byDictionaryOrder)
sort(["a","b","c"], byUnicodeOrder)
max(["a","b","c"], byDictionaryOrder)
//...
```


In that sense,  comparison is like a class, a trait. And every type can specify one or more ways to implement this trait, but it has nothing to do with sorting per se, although sort uses it

Reply:
Ah, now I get you. Would you be fine with having them in a seperate package? I would have thought of "sorting" as "everything that you need to sort your data", but that might be too big :) We could do `@dcos/sort` and `@dcos/comparators` or maybe put the sort part into a Util or something the like. I would like to have them co-located a bit more, but I would be fine with this change